### PR TITLE
Fix Max for Infinities

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: symengine/functions.cpp [hunk 1]
```diff
@@ -2799,6 +2799,11 @@ RCP<const Basic> max(const vec_basic &arg)
                 max_number = rcp_static_cast<const Number>(p);
 
             } else {
+                if (eq(*p, *Inf)) {
+                    return Inf;
+                } else if (eq(*p, *NegInf)) {
```
Comment (abhinavagarwal07) [inline symengine/functions.cpp]: will `max({NegInf, NegInf, NegInf})` work ?

Comment (isuruf) [inline symengine/functions.cpp]: Good point

Comment (isuruf) [inline symengine/functions.cpp]: Yes, it works, because of the if condition in line 2798

**Comment by abhinavagarwal07** [comment]
+1

**Pull request merged**